### PR TITLE
Cart and Wishlist

### PR DESCRIPTION
**Pull request by xsaurabhsuthar**
* Cart and wishlist functionalities added like add to cart, remove from cart, increment/decrement, add to wishlist and remove to wishlist.
* Login authentication added.

### CODE REVIEW

File: src/components/Cart/CartPriceCard/CartPriceCard.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          <span className="subtitle1__typography">Rs {totalActualPrice}</span>
+        </div>
+        <div className="price--item">
+          <span className="subtitle1__typography">Discount</span>
+          <span className="subtitle1__typography">-Rs {saved}</span>
+        </div>
+        {/* <div className="price--item">
+          <span className="subtitle1__typography">Delivery Charges</span>
+          <span className="subtitle1__typography">Rs99</span>
+        </div> */}
```
Comment (znjs) [inline src/components/Cart/CartPriceCard/CartPriceCard.jsx:23]: Can remove comments

File: src/reducer/reducer.js [hunk 2]
```diff
@@ -61,103 +61,10 @@ export const reducer = (state, action) => {
...
-        wishlist: {
-          ...state.wishlist,
-          products: [
-            ...state.wishlist.products.filter(
-              (product) => product._id !== action.payload._id
-            ),
-          ],
-        },
-      };
+    case "SET_CART":
```
Comment (znjs) [inline src/reducer/reducer.js:64]: Can use a variable instead of string to avoid error by typos

File: src/reducer/reducer.js [hunk 3]
```diff
@@ -61,103 +61,10 @@ export const reducer = (state, action) => {
...
+      return { ...state, cart: action.payload };
+    case "SET_WISHLIST":
```
Comment (znjs) [inline src/reducer/reducer.js:66]: Can use a variable instead of string to avoid error by typos

File: src/routes/Cart/Cart.css [hunk 4]
```diff
@@ -0,0 +1,103 @@
+.cart__section {
+  display: flex;
+  flex-direction: column;
+  /* flex-wrap: wrap; */
+  gap: 2rem;
+  /* height: 100%; */
```
Comment (znjs) [inline src/routes/Cart/Cart.css:6]: Can remove the comments

File: src/routes/Cart/Cart.css [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  overflow-y: auto;
+  align-items: center;
+}
+
+/* Wishlist cards */
+.wishlist--cards {
+  display: flex;
+  flex-direction: column;
+  gap: 2rem;
+  /* overflow-y: auto; */
```
Comment (znjs) [inline src/routes/Cart/Cart.css:16]: Can remove the comments

File: src/routes/Cart/Cart.jsx [hunk 6]
```diff
@@ -1,5 +1,33 @@
+// import axios from "axios";
 import React from "react";
+import { CartPriceCard } from "../../components/";
+import { EmptyCart } from "../../components/";
+import { ProductCartCard } from "../../components/";
```
Comment (znjs) [inline src/routes/Cart/Cart.jsx:5]: Can import all components in one line

**Review by znjs — APPROVED** [review]
Good folder structure
Good naming conventions used

File: src/components/Cart/CartPriceCard/CartPriceCard.jsx (hunk 1 above)
Comment (xsaurabhsuthar) [reply src/components/Cart/CartPriceCard/CartPriceCard.jsx:23]: Ok, will remove it.

File: src/reducer/reducer.js (hunk 2 above)
Comment (xsaurabhsuthar) [reply src/reducer/reducer.js:64]: Will update this.

File: src/routes/Cart/Cart.css (hunk 4 above)
Comment (xsaurabhsuthar) [reply src/routes/Cart/Cart.css:6]: Ok.

File: src/routes/Cart/Cart.css (hunk 5 above)
Comment (xsaurabhsuthar) [reply src/routes/Cart/Cart.css:16]: sure.

**Comment by xsaurabhsuthar** [comment]
Thanks @znjs  for the review.

**Review by ScoobyDubeyDo — APPROVED** [review]
Clean code 
Good file Structure 
Good naming

**Pull request merged**